ci/run/bazel_sharding/tests/test_bazel_sharding.py

```diff
@@ -0,0 +1,138 @@
+import pytest
+import os
+import shutil
+import sys
+import tempfile
+
+# Required for bazel
+file_parent = os.path.dirname(__file__)
+sys.path.append(os.path.join(file_parent, "../"))
+
+import bazel_sharding  # noqa: E402
+
+WORKSPACE_KEY = "workspace"
+
+
+def _prefix_rules(rules):
+    return list({f"//{WORKSPACE_KEY}:{rule}" for rule in rules})
+
+
+size_rules = _prefix_rules(["test_small", "test_medium", "test_large", "test_enormous"])
+timeout_rules = _prefix_rules(
+    ["test_short", "test_moderate", "test_long", "test_eternal"]
+)
+size_and_timeout_rules = _prefix_rules(["test_both_size_and_timeout"])
+manual_rules = _prefix_rules(["test_default"])
+all_rules = size_rules + timeout_rules + manual_rules + size_and_timeout_rules
+
+
+@pytest.fixture
+def mock_build_dir():
+    tmpdir = tempfile.mkdtemp()
+    with open(os.path.join(tmpdir, "WORKSPACE"), "w") as f:
+        f.write(
+            """
+workspace(name = "fake_workspace")
+            """
+        )
+    os.makedirs(os.path.join(tmpdir, WORKSPACE_KEY), exist_ok=True)
+    shutil.copyfile(
+        os.path.join(file_parent, "mock_BUILD"),
+        os.path.join(tmpdir, WORKSPACE_KEY, "BUILD"),
+    )
+    cwd = os.getcwd()
+    os.chdir(os.path.join(tmpdir, WORKSPACE_KEY))
+    yield
+    os.chdir(cwd)
+    shutil.rmtree(tmpdir, ignore_errors=True)
+
+
+def test_bazel_sharding_end_to_end(mock_build_dir):
```

Comment:
Can you add docstrings to all these tests explaining what the test is doing and what we assert?